### ISSUE TITLE
Duplicate recurring transfer checks

### ISSUE THREAD

**Issue by MuthuVega** [vegaprotocol/vega#6099]
Currently duplicate recurring transfers are checked based on the 2 fields - from and to accounts.
Think we need to add transfer asset in the duplicate checking as the above check seems to be very rigid and as the below set up fails.

Trying to set up the below 2 recurring transfers and the second one fails. 

**Transfers:**

```
{"id":"76105-38","block":"ECEB1D1956D953CBB900EC2436004F813C42628FA17EFF266858DA8ABF292AA5","type":"BUS_EVENT_TYPE_TRANSFER","transfer":{"id":"0cf7c339f6a1bfee70dc0915fd6a7045db405071d65e1b4f241f0d06fdd0f21c","from":"da4a6927a7ad300a306d184b0f702d327fe16f32d40cd339d85d41061c11cec7","fromAccountType":"ACCOUNT_TYPE_GENERAL","to":"4b94c941fd29b1f6ac3cb0fb92c917d874642f5cd839342562a5ce10ce8d808c","toAccountType":"ACCOUNT_TYPE_REWARD_MARKET_PROPOSERS","asset":"4b94c941fd29b1f6ac3cb0fb92c917d874642f5cd839342562a5ce10ce8d808c","amount":"100000000000000000000","reference":"TestTransfer","status":"STATUS_PENDING","timestamp":"1662625679701015000","recurring":{"startEpoch":"9121","factor":"1","dispatchStrategy":{"assetForMetric":"4b94c941fd29b1f6ac3cb0fb92c917d874642f5cd839342562a5ce10ce8d808c","metric":"DISPATCH_METRIC_MARKET_VALUE"}}},"version":1,"chainId":"test-chain-AFFKi2","txHash":"4B682B530EFF2E8E859A24DAC7994319C85EF3B1196856CBBD966C7152194155"}
{"id":"76141-38","block":"5277FF1855A00D9012A9417F8A6FC5F790EE5AF6040BA88DD65946EFF7506292","type":"BUS_EVENT_TYPE_TRANSFER","transfer":{"id":"704c610cca4b8743892506415dcce41dd03caf52579b2e8091a46464d1a1173e","from":"da4a6927a7ad300a306d184b0f702d327fe16f32d40cd339d85d41061c11cec7","fromAccountType":"ACCOUNT_TYPE_GENERAL","to":"4b94c941fd29b1f6ac3cb0fb92c917d874642f5cd839342562a5ce10ce8d808c","toAccountType":"ACCOUNT_TYPE_REWARD_MARKET_PROPOSERS","asset":"b4f2726571fbe8e33b442dc92ed2d7f0d810e21835b7371a7915a365f07ccd9b","amount":"20000000000000000000","reference":"TestTransfer","status":"STATUS_REJECTED","timestamp":"1662625711066581000","recurring":{"startEpoch":"9122","factor":"1","dispatchStrategy":{"assetForMetric":"4b94c941fd29b1f6ac3cb0fb92c917d874642f5cd839342562a5ce10ce8d808c","metric":"DISPATCH_METRIC_MARKET_VALUE"}}},"version":1,"chainId":"test-chain-AFFKi2","txHash":"E91A6CCE24BCEC56F2A9A566C925EF312E9EBEFE696F0EB8DBA080DBBAE269D5"}

```

**Errors:**

`{"id":"76141-39","block":"5277FF1855A00D9012A9417F8A6FC5F790EE5AF6040BA88DD65946EFF7506292","type":"BUS_EVENT_TYPE_TX_ERROR","txErrEvent":{"partyId":"da4a6927a7ad300a306d184b0f702d327fe16f32d40cd339d85d41061c11cec7","errMsg":"Transfer Funds - cannot submit duplicate recurring transfer with same from and to","transfer":{"fromAccountType":"ACCOUNT_TYPE_GENERAL","to":"4b94c941fd29b1f6ac3cb0fb92c917d874642f5cd839342562a5ce10ce8d808c","toAccountType":"ACCOUNT_TYPE_REWARD_MARKET_PROPOSERS","asset":"b4f2726571fbe8e33b442dc92ed2d7f0d810e21835b7371a7915a365f07ccd9b","amount":"20000000000000000000","reference":"TestTransfer","recurring":{"startEpoch":"9122","factor":"1","dispatchStrategy":{"assetForMetric":"4b94c941fd29b1f6ac3cb0fb92c917d874642f5cd839342562a5ce10ce8d808c","metric":"DISPATCH_METRIC_MARKET_VALUE"}}}},"version":1,"chainId":"test-chain-AFFKi2","txHash":"E91A6CCE24BCEC56F2A9A566C925EF312E9EBEFE696F0EB8DBA080DBBAE269D5"}
`

**Comment by MuthuVega** [comment]
Please let me know if you need more info

**Comment by ze97286** [comment]
Closed by #6098
